griptape/tasks/tool_task.py

```diff
@@ -84,7 +84,11 @@ def run(self) -> BaseArtifact:
             subtask.after_run()
 
             if isinstance(subtask.output, ListArtifact):
-                self.output = subtask.output[0]
+                first_artifact = subtask.output[0]
+                if isinstance(first_artifact, BaseArtifact):
+                    self.output = first_artifact
+                else:
+                    raise ValueError(f"Output is not an Artifact: {type(first_artifact)}")
```

Comment:
Why would this happen? Did you add this after observing an issue?

Reply:
This was introduced by adding generics to `ListArtifact`. Changed this to satisfy pyright.